File: Java/Shortest_Word.java
public class Kata {
    public static int findShort(String s) {
        String[] words = s.split(" ");
        int min = 2147483647;
        for (String word : words)
          min = (word.length() < min ? word.length() : min);
        return min;
    }
}

File: C/Shortest_Word.c
#include <sys/types.h>
#include <string.h>
#include <limits.h>

ssize_t find_short(const char *s)
{
    ssize_t min = SSIZE_MAX;
    int chars = 0;
    
    for (int i = 0; s[i]; i++) {
        if (s[i] == ' ') {
            if (chars > 0)
                min = (chars < min) ? chars : min;
            chars = 0;
        } else chars++;
    }
    if (chars > 0) min = (chars < min) ? chars : min;
  return (min);
}
